pkg/controller/kfservice/kfservice_controller.go

```diff
@@ -130,16 +137,26 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 			// For additional cleanup logic use finalizers.
 			return reconcile.Result{}, nil
 		}
+		return reconcile.Result{}, err
+	}
+
+	configMap := &v1.ConfigMap{}
+	err := r.Get(context.TODO(), types.NamespacedName{Name: constants.KFServiceConfigMapName, Namespace: constants.KFServingNamespace}, configMap)
```

Comment:
so do we load and initialize the configmap on every reconcile call? is it not possible to cache the config map and only update if we get an update that says it has changed?

Reply:
This is done by `controller-runtime` where internally maintains a cache and wire up updates from upstream. Here we are just getting the object from cache, this is the API we are calling under the hood.
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/internal/cache_reader.go#L49

Reply:
sweet!